src/API.jl

```diff
@@ -1050,4 +1050,31 @@ Package(name::AbstractString) = PackageSpec(name)
 Package(name::AbstractString, uuid::UUID) = PackageSpec(name, uuid)
 Package(name::AbstractString, uuid::UUID, version::VersionTypes) = PackageSpec(name, uuid, version)
 
+
+"""
+    get_uuid(m::Module)
+
+Given a `Module`, find the `UUID` of the parent package to that `Module`.  If none can be
+found (e.g. `Main`, `Base`, anonymous modules, etc...) returns `nothing`.
+"""
+function get_uuid(m::Module)
+    return Base.PkgId(m).uuid
+end
+
+"""
+    get_version(pkg)
+
+Given a `UUID` identifying a package, returns the version of that package within the
+current project.  If the package cannot be found by the given UUID, throws an
+`ArgumentError`.
+"""
+function get_version(uuid::UUID)
```

Comment:
This is not always correct. I don't think we can support this until the version number of the loaded package is recorded in base.

Reply:
We at least likely want this to be a precompiled query and not a runtime one (for relocatability).

Reply:
> This is not always correct. 

How would this not be correct?  `ctx.env.manifest` would change as the user does `activate` and such, right?

> We at least likely want this to be a precompiled query and not a runtime one (for relocatability).

If a user does `const version = Pkg.API.get_version(Pkg.API.get_uuid(@__MODULE__))` at the top level of their package, that should work, right?

Reply:
> ctx.env.manifest would change as the user does activate and such, right?

Yea but that is the point, it can change after you have already loaded the package.

Reply:
Ah, I see.  Perhaps that is solvable by documenting to users that they should just store the results at compile time?

Reply:
The primary usecase I have for this is for users easily getting the version of their own package, yes.  It's brittle to hardcode it in the source; people forget to update it, and it seems wasteful to import an entire TOML parser to try and figure out the UUID/version from the `Project.toml`.